File: Medium Problems/Leetcode 34.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int first = -1;
        int last = -1;
        int n = nums.size();
        
        if(n == 0){
            return {first,last};
        }

        for(int i = 0; i < n; i++){
            if(nums[i] == target){
                first = i;
                break;
            }
            if(nums[i] > target)
                break;
        }
        if(first != -1){
            for(int i = first; i < n; i++){
                if(nums[i] == target){
                    last = i;
                }
                if(nums[i] > target)
                    break; 
            }
        }
        return {first,last};
    }
};

File: Medium Problems/Leetcode 34.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def find_first(nums):
            first = -1
            l = 0
            r = len(nums) - 1
            while l <= r:
                mid = l + (r - l) // 2
                if nums[mid] == target:
                    first = mid
                    r = mid - 1
                elif nums[mid] > target:
                    r = mid - 1
                else:
                    l = mid + 1
            return first

        def find_last(nums):
            last = -1

            l = 0
            r = len(nums) - 1

            while l <= r:
                mid = l + (r - l) // 2
                if nums[mid] == target:
                    last = mid
                    l = mid + 1
                elif nums[mid] > target:
                    r = mid - 1
                else:
                    l = mid + 1
            return last
        if not nums:
            return -1, -1
        return find_first(nums), find_last(nums)
